misk-metrics-digester/src/test/kotlin/misk/metrics/digester/RegistryTest.kt

```diff
@@ -0,0 +1,59 @@
+package misk.metrics.digester
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class RegistryTest {
+
+  @Test
+  fun registyDigest() {
+    val registry = histogramRegistryFactory()
+
+    val histogram = registry.newHistogram("name", "help", listOf(), mapOf(0.1 to 0.1, 0.2 to 0.2))
+    val metric = histogram.labels()
+    metric.observe(1.23)
+    metric.observe(4.56)
+
+    val firstWindow = histogram.getMetric().digest.windows[0].digest as FakeDigest
+    assertThat(firstWindow.addedValues).isEqualTo(listOf(1.23, 4.56))
+    assertThat(histogram.count()).isEqualTo(3)
+  }
+
+  @Test
+  fun registryDigestWithLabels() {
+    val registry = histogramRegistryFactory()
+
+    val histogram = registry.newHistogram("name", "help", listOf("label1", "label2"),
+        mapOf(0.1 to 0.1, 0.2 to 0.2))
+
+    //Check that passed labels exist
+    assertThat(histogram.getMetric("label1")).isNotNull()
```

Comment:
This should not pass. The code should throw an user-friendly exception, as the user must pass a label value for each label name.

`histogram.getMetric()` and `histogram.getMetric("a","b","c","d")` should also throw.